sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java

```diff
@@ -175,15 +193,20 @@ public AutoConfiguredOpenTelemetrySdk build() {
       }
     }
 
+    SdkTracerProviderBuilder tracerProviderBuilder = SdkTracerProvider.builder();
     ConfigProperties config = getConfig();
+    tracerProviderBuilder = tracerProviderCustomizer.apply(tracerProviderBuilder, config);
+
+    // TODO - with the builder customizer, the user may set stuff that is later overridden by us
     Resource resource = ResourceConfiguration.configureResource(config, resourceCustomizer);
+    tracerProviderBuilder.setResource(resource);
```

Comment:
The issue here is that exposing the builder, allows the consumer to set the `Resource`, that we override with our own lookup. This also happens with other stuff.

Should we also add a check if things we set into the builder directly, don't override them? I think it makes sense.

Reply:
@radcortez Sorry for taking so long to get back to you. We already have an SPI for tracerprovider customization that has the same issue. Let's not worry about it for now as it's a small corner case, and we can address if everyone finds it as a bug. We just want this builder to have the same functionality as our current SPI